src/dataflow/src/source/s3.rs

```diff
@@ -612,12 +677,14 @@ async fn download_object(
         }
 
         Some(DownloadMetricUpdate {
-            bytes: buf.len() as u64,
+            bytes: bytes_read,
             messages,
             sent,
         })
     } else {
-        log::warn!("get object response had no body");
+        if let Err(e) = tx.send(Err(anyhow!("Get object response had no body: {}", key))) {
+            log::debug!("unable to send error on stream: {}", e);
+        }
```

Comment:
Hm so this code is currently annoyingly factored to handle tx send errors. We should try to shut down all tasks as soon as any send fails.

It shouldn't be part of this PR, but at some point we should refactor this function to return a custom enum (instead of Option) that looks like:

```rust
enum DataUpdate {
    Updated(DownloadMetricUpdate),
    NoUpdate(Sent),
}
```